### PR TITLE
Fix Spoofchecker callmaps

### PR DESCRIPTION
**Pull request by othercorey**
fixes https://github.com/vimeo/psalm/issues/9254

### CODE REVIEW

**Comment by othercorey** [comment]
setRestrictionLevel() does not exist in the docs, but it a defined method. 

https://github.com/php/doc-en/issues/2283

**Comment by weirdan** [comment]
Thanks!

**Comment by stof** [comment]
> setRestrictionLevel() does not exist in the docs, but it a defined method.

It is a _conditional_ method. It exists only when compiling ext-intl with ICU 58+. Not sure how this gets represented in psalm.

File: dictionaries/CallMap.php [hunk 1]
```diff
@@ -13471,11 +13471,11 @@
 'SplTempFileObject::valid' => ['bool'],
 'SplType::__construct' => ['void', 'initial_value='=>'mixed', 'strict='=>'bool'],
 'Spoofchecker::__construct' => ['void'],
-'Spoofchecker::areConfusable' => ['bool', 's1'=>'string', 's2'=>'string', '&w_error='=>'string'],
-'Spoofchecker::isSuspicious' => ['bool', 'text'=>'string', '&w_error='=>'string'],
-'Spoofchecker::setAllowedLocales' => ['void', 'locale_list'=>'string'],
-'Spoofchecker::setChecks' => ['void', 'checks'=>'long'],
-'Spoofchecker::setRestrictionLevel' => ['void', 'restriction_level'=>'int'],
+'Spoofchecker::areConfusable' => ['bool', 'string1'=>'string', 'string2'=>'string', '&w_errorCode='=>'int'],
+'Spoofchecker::isSuspicious' => ['bool', 'string'=>'string', '&w_errorCode='=>'int'],
```
Comment (stof) [inline dictionaries/CallMap.php:13475]: the name is `errorCode` in the PHP stubs, not `w_errorCode`

Comment (othercorey) [reply dictionaries/CallMap.php:13475]: This is an internal psalm annotation. The callmap is verified against reflection so it matches PHP 8.1 reflection.

**Pull request merged**